com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs

```diff
@@ -472,12 +552,12 @@ private void SendSceneEventData(ulong[] targetClientIds)
             }
             var message = new SceneEventMessage
             {
-                EventData = SceneEventData
+                EventData = SceneEventDataPool[sceneEventDataIndex]
             };
             var size = m_NetworkManager.SendMessage(message, k_DeliveryType, targetClientIds);
 
             m_NetworkManager.NetworkMetrics.TrackSceneEventSent(
-                targetClientIds, (uint)SceneEventData.SceneEventType, ScenesInBuild[(int)SceneEventData.SceneIndex], size);
+                targetClientIds, (uint)SceneEventDataPool[sceneEventDataIndex].SceneEventType, ScenesInBuild[(int)SceneEventDataPool[sceneEventDataIndex].SceneIndex], size);
```

Comment:
This was already present in the code before the change, but I'm curious. How come the typecast is not an indication of a potential improvement? Shouldn't `TrackSceneEventSent` already take a parameter of the type of `SceneEventData.SceneEventType` ? It seem it would be safer if both were the same type.

Reply:
Good question, that perhaps the tools team would be better suited to answer

Reply:
`INetworkMetrics` is type agnostic. Probably because we don't want to leak types out externally.